multilinear-util/src/point.rs

```diff
@@ -0,0 +1,655 @@
+use core::ops::{Index, Range};
+
+use p3_field::Field;
+use rand::Rng;
+use rand::distr::{Distribution, StandardUniform};
+
+/// A point `(x_1, ..., x_n)` in `F^n` for some field `F`.
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct MultilinearPoint<F>(pub Vec<F>);
```

Comment:
Hmm, as we update things in the p3_whir repo, do you think you can remove the `pub` on the inner argument? (Probably leave it pub(crate) though as it would be painful otherwise).

Will need to add a new method but that should be easy.

```suggestion
pub struct MultilinearPoint<F>(pub(crate) Vec<F>);
```